src/formatter/terminal_themed.rs

```diff
@@ -0,0 +1,76 @@
+use std::cell::RefCell;
+use std::io::Write;
+use std::sync::Arc;
+use termcolor::{Color, ColorChoice, ColorSpec, StandardStream, WriteColor};
+
+use super::{Formatter, HighlightEvent, Theme};
+use crate::constants::HIGHLIGHT_NAMES;
+use crate::Result;
+
+/// A formatter for highlighting into terminal output with ANSI color codes.
+///
+/// - The primary color of each style is used for the text color.
+/// - The secondary color is not used in this formatter.
+///
+/// This formatter holds theme data in an [`Arc<Theme>`], making it cheap to clone and usable
+/// from multiple threads.
+#[derive(Debug)]
+pub struct TerminalFormatter {
+    theme: Arc<Theme>,
+    stdout: RefCell<StandardStream>,
```

Comment:
`RefCell<T>` isn't `Send` or `Sync` (that is, multi-thread-compatible) like `Arc<T>` is.

For something like this, you *probably* don't want to be writing to standard output/error from multiple threads anyways, so I would just do something like:

```rs
#[derive(Debug, Clone)]
pub struct Terminal {
    theme: Theme,
    stream: StandardSteam
}
```